splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1124,7 +1124,7 @@ public boolean canWriteCache(TransactionController xactMgr) throws StandardExcep
         DDLWatcher ddlWatcher=driver.ddlWatcher();
         if(xactMgr==null)
             xactMgr = getTransactionCompile();
-        return ddlWatcher.canWriteCache((TransactionManager)xactMgr);
+        return xactMgr != null && ddlWatcher.canWriteCache((TransactionManager)xactMgr);
```

Comment:
Is this necessary to fix the problem or is it just preventing a similar NPE in the future?

Reply:
see above